docs/launch-behavior.md

```diff
@@ -0,0 +1,137 @@
+| `atom --new-window /a`      | [] [/a]      | [] [/a]          | [] [/a]      |
+
+### Open window, project root
+
+With the following starting state:
+
+> [/a]
+
+| Action | Outcome (<=1.35.1) | Outcome (1.36.0) | Outcome (1.36.1) |
+| --- | --- | --- | --- |
+| `atom /a/1.md`              | [/a 1.md]      | [/a 1.md]          | [/a 1.md]    |
+| `atom /a`                   | [/a]           | [/a]               | [/a]         |
+| `atom /b/3.md`              | [/a 3.md] | [/a] [ 3.md] :new: | [/a] [ 3.md] |
+| `atom /b`                   | [/a] [/b]      | [/a] [/b]          | [/a] [/b]    |
+| `atom --add /a/1.md`        | [/a 1.md]      | [/a 1.md]          | [/a 1.md]    |
+| `atom --add /a`             | [/a]           | [/a]               | [/a]         |
+| `atom --add /b/3.md`        | [/a,/b 3.md]   | [/a 3.md] :new:    | [/a 3.md]    |
+| `atom --add /b`             | [/a,/b]        | [/a,/b]            | [/a,/b]      |
+| `atom --new-window /a/1.md` | [/a] [/a 1.md] | [/a] [ 1.md] :new: | [/a] [ 1.md] |
+| `atom --new-window /a`      | [/a] [/a]      | [/a] [/a]          | [/a] [/a]    |
+| `atom --new-window /b/3.md` | [/a] [/b 3.md] | [/a] [ 3.md] :new: | [/a] [ 3.md] |
+| `atom --new-window /b`      | [/a] [/b]      | [/a] [/b]          | [/a] [/b]    |
+
+### Open windows, one with a project root and one without
+
+> [/a] []
+
+| Action | Outcome (<=1.35.1) | Outcome (1.36.0) | Outcome (1.36.1) |
+| --- | --- | --- | --- |
+| `atom /a/1.md`              | [/a 1.md] []      | [/a 1.md] []          | [/a 1.md] []       |
+| `atom /a`                   | [/a] []           | [/a] []               | [/a] []            |
+| `atom /b/3.md`              | [/a] [/b 3.md]    | [/a] [] [ 3.md] :new: | [/a] [ 3.md] :new: |
+| `atom /b`                   | [/a] [/b]         | [/a] [] [/b]          | [/a] [] [/b]       |
+| `atom --add /a/1.md`        | [/a 1.md] []      | [/a 1.md] []          | [/a 1.md] []       |
+| `atom --add /a`             | [/a] []           | [/a] []               | [/a] []            |
+| `atom --add /b/3.md`        | [/a] [/b 3.md]    | [/a] [ 3.md] :new:    | [/a] [ 3.md]       |
+| `atom --add /b`             | [/a] [/b]         | [/a] [/b]             | [/a] [/b]          |
+| `atom --new-window /a/1.md` | [/a] [] [/a 1.md] | [/a] [] [ 1.md] :new: | [/a] [] [1.md]     |
+| `atom --new-window /a`      | [/a] [] [/a]      | [/a] [] [/a]          | [/a] [] [/a]       |
+| `atom --new-window /b/3.md` | [/a] [] [/b 3.md] | [/a] [] [ 3.md] :new: | [/a] [] [ 3.md]    |
+| `atom --new-window /b`      | [/a] [] [/b]      | [/a] [] [/b]          | [/a] [] [/b]       |
+
+## File manager integration actions
+
+### No open windows
+
+When no Atom windows are open, file manager context operations have the following outcomes:
+
+| Action | Outcome (<=1.35.1) | Outcome (1.36.0) | Outcome (1.36.1) |
+| --- | --- | --- | --- |
+| Click on `/a/1.md` | [/a 1.md] | [ 1.md] :new: | [ 1.md] |
```

Comment:
I think this change is probably OK. However, it is a _change_, and this change wasn't an explicit goal of #18608, so I think we should be open to restoring the previous behavior if there's a compelling argument for doing so.

Reply:
Yeah, I'm on the fence about this one. It feels like we'd want to avoid indexing the directory here too, so that we launch quickly when clicking on a file in your home directory just as we do when you run `atom ~/file.txt` from the CLI. But I could see an argument that user expectations may differ?

Reply:
> But I could see an argument that user expectations may differ?

I don't personally anticipate user expectations differing here. I personally prefer this change, but I'll be curious to see if it gets any pushback.